examples/cvode/parallel/cvAdvDiff_diag_p.c

```diff
@@ -123,6 +124,11 @@ int main(int argc, char *argv[])
   retval = SUNContext_Create((void*) &comm, &sunctx);
   if(check_retval(&retval, "SUNContext_Create", 1, my_pe)) MPI_Abort(comm, 1);
 
+  /* Setup different error handler stack so that we abort after logging */
```

Comment:
I'm not sure that I understand this comment.  Is the idea that by replacing the existing error handler stack with `NULL`, then any error will immediately result in an abort?  If so, I'd rephrase this so that it isn't focused on "logging" but on catching errors.

Also, it looks like this 3-call sequence is used in many examples.  Is this recommended for users?  Perhaps we should create a utility routine that will just do the three of these all together, and that has a descriptive name for the intended use case, e.g., `SUNContext_ForceAbortOnErrors`?

Reply:
This is only done in this example as a demonstration of how you can set different error handlers. A `SUNContext_ForceAbortOnErrors` shortcut isn't a bad idea though.

Reply:
At first glance of the PR, you may see many examples changed, this is misleading. Most of the examples did not change with this PR. The ones that did are a tiny change to use `SUNLsStatus` instead of `int` where appropriate and then this example referenced above to demonstrate the error handler stack. The bulk of the changes in this PR are in the `src` directory.